### PR TITLE
Use Webpack

### LINKED ISSUE #252: Use a bundler
Currently the release tarball is exceptionally big because you're shipping `node_modules`. With the help of a bundler, third party scripts and other resources can be combined into a few files. Then the app tarball should shrink significantly again.

I would suggest looking into Webpack, as we have some experience with it in other apps and the Nextcloud server.

### PR DESCRIPTION
**Pull request by gary-kim**
Closes #252 

Being a bit conservative with the milestone for now. :smile: 

There's quite a lot more that could be done for cleanup but I think it'd be better to put that in a follow-up PR.

EDIT: Someone who has access to settings for the repo: could you enable the `projects` feature? I'd like to make a project for migration to Vue.
EDIT 3: Once this is done, PR incoming for migrating admin settings to Vue.
EDIT 4: 
@eneiluj @jancborchardt `package-lock.json` and `composer.lock` being in the `.gitignore` file caused me some confusion. I just read #44.
I just removed those lines from `.gitignore` and committed those files in this PR.
Just for future reference, `package-lock.json` and `composer.lock` should be committed into source control: [package-lock.json documentation](https://docs.npmjs.com/files/package-lock.json) && [composer.lock documentation](https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control) :smile:. (I don't mean that with any negative intent)

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Just for future reference, `package-lock.json` and `composer.lock` should be committed into source control: [package-lock.json documentation](https://docs.npmjs.com/files/package-lock.json) && [composer.lock documentation](https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control) smile. (I don't mean that with any negative intent)

Yes!

**Comment by ChristophWurst** [comment]
> +7,999

Are you trying to sell the PR? :P

File: package.json [hunk 1]
```diff
@@ -24,24 +24,48 @@
...
-    "opening_hours": "^3.5.0"
+    "opening_hours": "^3.5.0",
+    "ua-parser-js": "^0.7.20"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.8.4",
+    "@babel/plugin-syntax-dynamic-import": "^7.8.3",
+    "@babel/preset-env": "^7.8.4",
+    "@nextcloud/browserslist-config": "^1.0.0",
+    "@nextcloud/router": "^1.0.0",
```
Comment (ChristophWurst) [inline package.json]: is it a not a prod dependency? :thinking:

**Comment by tacruc** [comment]
How does this work together with #217? 
As I see both PRs add some webpack.prod.js for example.
@paulschwoerer Cloud you review, too.

**Comment by gary-kim** [comment]
Huh, interesting. This is going to be a messy merge. #217 set up a lot of the same things as this PR with the main difference being that #217, the setup was just for the new feature and here, the idea is to migrate all of the JS for the app.

**Comment by tacruc** [comment]
Thats what I was afraid of. What would be the besteht order of merging them?
@paulschwoerer

File: package.json (hunk 1 above)
Comment (paulschwoerer) [reply package.json]: I agree, this looks like it should be a prod dependency.

**Comment by paulschwoerer** [comment]
@gary-kim Solid work!
It should actually be quite straightforward to merge this with the changes I did. I'd suggest to merge this commit first and I'll take care about merging the changes into #217 before merging #217 into master.

I was also interested in migrating the logged in part to Vue, I'd be glad to lend a hand in doing so. In my PR you might find some useful components, that can be expanded on.

File: package.json (hunk 1 above)
Comment (gary-kim) [reply package.json]: Done! Thank you for your review!

**Comment by gary-kim** [comment]
> @gary-kim Solid work!

You too @paulschwoerer!



> In my PR you might find some useful components, that can be expanded on.

Thank you! I'll be sure to make good use of those.

**Comment by tacruc** [comment]
Is it ready to merge?

**Comment by gary-kim** [comment]
> Is it ready to merge?

:rocket: Ready to merge!

**Pull request merged**